src/node_trace_events.cc

```diff
@@ -63,6 +63,9 @@ void NodeCategorySet::New(const FunctionCallbackInfo<Value>& args) {
     if (!*val) return;
     categories.emplace(*val);
   }
+  // Starts the Tracing Agent if it wasn't started already (e.g. through
+  // a command line flag.)
+  StartTracingAgent();
```

Comment:
Should this happen when the module is being loaded, rather than in the constructor? Does that make a difference?

Reply:
I feel like it doesn't make a huge difference since I'd expect the module load and first constructor call to be in close proximity to each other. That being said my thinking here is it might be better to do it lazily (so, in the constructor)? Since the user-facing API is called `createTracing` which is more suggestive of an explicit internal change than simply requiring the module. In addition, `async_hooks` also doesn't really enable certain internals unless a hook is first enabled. (Though, that might suggest that `StartTracingAgent` belongs even later, in `Enable`.)

Reply:
> Though, that might suggest that `StartTracingAgent` belongs even later, in `Enable` -- what do you think?

Yeah, it sounds like that’s what we want if we’re going for lazy enabling?

Reply:
Sounds good, I've moved the `StartTracingAgent` call.